app/webpacker/components/MyCompetitions/UpcomingCompetitionTable.jsx

```diff
@@ -0,0 +1,109 @@
+import {
+  Icon,
+  Popup, Table, TableBody, TableHeader,
+} from 'semantic-ui-react';
+import React from 'react';
+import I18n from '../../lib/i18n';
+import { competitionStatusText } from '../../lib/utils/competition-table';
+import { competitionRegistrationsUrl, editCompetitionsUrl } from '../../lib/requests/routes.js.erb';
+import ReportTableCell from './ReportTableCell';
+import { countries } from '../../lib/wca-data.js.erb';
+
+const registrationStatusIcon = (registrationStatus) => {
+  switch (registrationStatus?.competing_status) {
+    case 'pending': return <Icon name="hourglass" />;
+    case 'waiting_list': return <Icon name="hourglass" />;
+    case 'accepted': return <Icon name="check circle" />;
+    case 'cancelled': return <Icon name="trash" />;
+    case 'rejected': return <Icon name="trash" />;
+    default: return <Icon />;
+  }
+};
+
+const competitionStatusIcon = (competition) => {
+  if (competition['registration_not_yet_opened?']) {
+    return <Icon name="clock" color="blue" />;
+  }
+  if (competition['registration_past?']) {
+    return <Icon name="user times" color="red" />;
+  }
+  if (competition['registration_full?']) {
+    return <Icon name="user clock" color="orange" />;
+  }
+  return <Icon name="user plus" color="green" />;
+};
+
+export default function UpcomingCompetitionTable({
+  competitions, permissions, registrationStatuses, shouldShowRegistrationStatus = false,
+}) {
+  return (
+    <Table>
+      <TableHeader>
+        <Table.Row>
+          { shouldShowRegistrationStatus && <Table.HeaderCell /> }
+          <Table.HeaderCell>
+            {I18n.t('competitions.adjacent_competitions.name')}
+          </Table.HeaderCell>
+          <Table.HeaderCell>
+            {I18n.t('competitions.adjacent_competitions.location')}
+          </Table.HeaderCell>
+          <Table.HeaderCell>
+            {I18n.t('competitions.adjacent_competitions.date')}
+          </Table.HeaderCell>
+          <Table.HeaderCell />
+          <Table.HeaderCell />
+          <Table.HeaderCell />
+          <Table.HeaderCell />
+          <Table.HeaderCell />
```

Comment:
Should some of these be absent if the user has no relevant permissions for any competitions?

Reply:
yeah, changed it